backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/DemoOktaRepository.java

```diff
@@ -30,49 +32,124 @@
         }
 
-        LOG.info("Done initializing Demo Okta service.");
+        orgRoleUsernamesMap.put(orgExternalId, roleUsernamesMap);
+    }
+
+    // Does not overwrite demo users who were created on initialization
+    public void createUser(IdentityAttributes userIdentity, Organization org) {
+        String organizationExternalId = org.getExternalId();
+        OrganizationRoleClaims orgRoles = new OrganizationRoleClaims(organizationExternalId, 
+                                                                     Set.of(OrganizationRole.USER));
+        usernameOrgRolesMap.putIfAbsent(userIdentity.getUsername(), orgRoles);
+
+        Map<OrganizationRole,List<String>> roleUsernamesMap = orgRoleUsernamesMap.get(organizationExternalId);
+        List<String> usernames = roleUsernamesMap.get(OrganizationRole.USER);
+        if (!usernames.contains(userIdentity.getUsername())) {
+            usernames.add(userIdentity.getUsername());
+            roleUsernamesMap.put(OrganizationRole.USER, usernames);
+            orgRoleUsernamesMap.put(organizationExternalId, roleUsernamesMap);
+        }
+    }
+
+    public void updateUser(String oldUsername, IdentityAttributes userIdentity) {
+        usernameOrgRolesMap.put(userIdentity.getUsername(), usernameOrgRolesMap.get(oldUsername));
+
+        for (String org : orgRoleUsernamesMap.keySet()) {
+            Map<OrganizationRole,List<String>> roleUsernamesMap = orgRoleUsernamesMap.get(org);
+            for (OrganizationRole role : roleUsernamesMap.keySet()) {
+                List<String> usernames = roleUsernamesMap.get(role);
+                if (usernames.remove(oldUsername)) {
+                    usernames.add(userIdentity.getUsername());
+                    roleUsernamesMap.put(role, usernames);
+                }
+            }
+            orgRoleUsernamesMap.put(org, roleUsernamesMap);
+        }
     }
 
-    // Will add to these in a PR soon.
-    public void createUser(IdentityAttributes userIdentity, String organizationExternalId) {}
+    public OrganizationRole updateUserRole(String username, OrganizationRole role) {
+        OrganizationRoleClaims oldRoleClaims = usernameOrgRolesMap.get(username);
+        Set<OrganizationRole> roles = new HashSet<>();
+        roles.add(OrganizationRole.USER);
+        roles.add(role);
+        OrganizationRoleClaims newRoleClaims = new OrganizationRoleClaims(oldRoleClaims.getOrganizationExternalId(),
+                                                                          roles);
+        usernameOrgRolesMap.put(username, newRoleClaims);
+        
+        for (String org : orgRoleUsernamesMap.keySet()) {
+            Map<OrganizationRole,List<String>> roleUsernamesMap = orgRoleUsernamesMap.get(org);
+            for (OrganizationRole r : roleUsernamesMap.keySet()) {
+                List<String> usernames = roleUsernamesMap.get(r);
+                if (r.equals(role)) {
+                    if (!usernames.contains(username)) {
```

Comment:
This seems like it kind of wants to be a `Set` rather than a `List`, but it would probably be a bad idea to actually care about efficiency on searches of our 5 demo users.